### ISSUE TITLE
Building with CMake 3.10 fails since  #285

### ISSUE THREAD

**Issue by shoshijak** [cp2k/dbcsr#289]
Building with CMake 3.10 fails since commit [34bfa005e1aba2982707917a499839aea220c764](https://github.com/cp2k/dbcsr/commit/34bfa005e1aba2982707917a499839aea220c764) (PR #285), with a segfault between configuration and generation:

```
-- FYPP preprocessor found.
Tests will run with 4 MPI ranks and 3 OpenMP threads each
-- Configuring done
Segmentation fault (core dumped)
```

All PRs subsequent to #285 thus fail the CI test `gnu-cmake-3.10.build`. 

Building with CMake >= 3.14 doesn't present any issues.

**Comment by dev-zero** [comment]
Can reproduce manually on Daint, but not locally, might be [this bug](https://gitlab.kitware.com/cmake/cmake/commit/ab3b549ee0dcd79607f8347a9b69c9b029590fbd).\
Reported to CSCS to forward it to SUSE.

**Comment by dev-zero** [comment]
OK, it's not only SUSEs CMake-3.10.2, but also the one in Ubuntu Bionic (and therefore in the `nvidia/cuda:devel`. Theo from CSCS sent me the following `Dockerfile` as a reproducer:

```
FROM nvidia/cuda:10.1-devel

RUN apt-get update && \
    apt-get install -y cmake git \
    libblas-dev liblapack-dev mpich python3-dev

RUN git clone --recursive https://github.com/cp2k/dbcsr.git && \
    mkdir build && cd build && \
    cmake -DCMAKE_CROSSCOMPILING_EMULATOR="" -DUSE_CUDA=ON -DUSE_CUBLAS=ON -DBLAS_FOUND=ON -DBLAS_LIBRARIES="-lsci_gnu_mpi_mp" -DLAPACK_FOUND=ON -DLAPACK_LIBRARIES="-lsci_gnu_mpi_mp" ../dbcsr/ --debug-output
```

So, we have 3 options:

1. Revert (parts of) the commit
2. Motivate distros to pull a backported patch for the issue
3. Drop support for CMake 3.10

Opinions?

**Comment by dev-zero** [comment]
oh, and removing `-DCMAKE_CROSSCOMPILING_EMULATOR=""` makes the segfault go away, so it is the cited CMake issue, meaning that before #285 we were not calling the emulator if needed, so the change in #285 is actually correct.

**Comment by shoshijak** [comment]
@dev-zero, thank you for doing all this research!

> 1. Revert (parts of) the commit

It's a good commit, and the code is correct, so I'd be reluctant to revert good code just because of a CMake issue.

> 2. Motivate distros to pull a backported patch for the issue

A lot of work, and not necessarily our job...
So I'd suggest we:

> 3. Drop support for CMake 3.10

and upgrade to CMake 3.12, so we can get rid of [these ugly lines](https://github.com/cp2k/dbcsr/blob/develop/CMakeLists.txt#L114-L117).

**Comment by alazzaro** [comment]
I agree that we should move to >=3.12. I'm doing it as part of my PR...

**Comment by alazzaro** [comment]
Closed via #307 (now we require CMake > 3.12)
